qutebrowser/browser/webkit/webkitelem.py

```diff
@@ -126,6 +126,14 @@ def set_value(self, value):
             value = javascript.string_escape(value)
             self._elem.evaluateJavaScript("this.value='{}'".format(value))
 
+    def caret_position(self):
+        self._check_vanished()
+        pos = self._elem.evaluateJavaScript('this.selectionStart')
+        assert isinstance(pos, (int, float, type(None)))
+        if pos is None:
+            return 0
+        return int(pos)
```

Comment:
As stated, it surprisingly is a float

Reply:
If we convert it to an `int` anyway, why'd you check the type first? Using
```python
pos = self._elem.evaluateJavaScript('this.selectionStart')
return int(pos or 0)
```
should be equivalent for most types, except it'll throw a ValueError instead of an AssertionError. Furthermore, it'll also accept other numerical types.

Reply:
I agree we can get rid of the `isinstance` check here, but I'd rather not do `int(pos or 0)` - I've seen too many funny things happen with JavaScript bridged to C++. If something goes wrong here, and Qt returns `{}` or something, I'd rather know about it than just assume 0.

Reply:
I'm sorry, I am having trouble understanding what is requested here. I gather the `isinstance` check can be removed, but at that point what is the difference between

```python
if pos is None:
    return 0
return int(pos)
```

and 

```python
return int(pos or 0)
```

?

Maybe I'm misunderstanding and I should in fact leave `isinstance` alone?

Reply:
The difference between the two is how e.g. `{}` is handled. I've seen QtWebEngine just return `{}` when accidentally passing some kind of JS object which isn't just made up of "primitive types".

If we do something like that, I'd much rather get an exception here than just silently assuming 0 (the reason I originally added those type checks is that I spent quite some time debugging a funny issue like that).

With

```python
if pos is None:
    return 0
return int(pos)
```

passing `{}` fails like expected, while with `return int(pos or 0)`, `0` is used.

Reply:
I see, it makes sense. I had tought that empty stuff would generally map to `None` but I held the wrong assumption.